executor/joiner.go

```diff
@@ -182,6 +200,34 @@ func (j *baseJoiner) filter(input, output *chunk.Chunk, outerColsLen int) (bool,
 	return chunk.CopySelectedJoinRows(input, innerColOffset, outerColOffset, j.selected, output)
 }
 
+// filterAndCheckOuterRowStatus is used to filter the result constructed by
+// tryToMatchOuters, the result is built by multiple outer rows and one inner
+// row. The returned outerRowStatusFlag slice value indicates the status of
+// each outer row (matched/unmatched/hasNull).
+func (j *baseJoiner) filterAndCheckOuterRowStatus(input, output *chunk.Chunk, innerColsLen int, outerRowStatus []outerRowStatusFlag) (_ []outerRowStatusFlag, _ error) {
```

Comment:
can we refactor it to call `j.Filter()` then update `outerRowStatus`?

Reply:
The code already did this?